### PR TITLE
feat: translate curation api for redesign

### PR DESCRIPTION
**Pull request by Bento007**
- #3616 

### Reviewers
**Functional:** 

**Readability:** 

---

## Changes
- add
- remove
- modify

### CODE REVIEW

File: tests/unit/backend/layers/common/base_api_test.py [hunk 1]
```diff
@@ -86,82 +88,93 @@ def _mock_assert_authorized_token(self, token: str, audience: str = None):
...
-    business_logic: BusinessLogic
-    crossref_provider: CrossrefProviderInterface # Can be mocked from the tests
-    uri_provider: UriProviderInterface
 
-    sample_dataset_metadata: DatasetMetadata
-    
+class NewBaseTest(BaseAuthAPITest):
     def setUp(self):
         super().setUp()
+        self.business_logic: BusinessLogic
```
Comment (ebezzi) [inline tests/unit/backend/layers/common/base_api_test.py:95]: These are just class member type hints. They shouldn't be in a method.

File: backend/layers/business/business.py [hunk 2]
```diff
@@ -128,6 +129,17 @@ def get_published_collection_version(self, collection_id: CollectionId) -> Optio
         """
         return self.database_provider.get_collection_mapped_version(collection_id)
 
+    def get_unplublished_collection_version_from_canonical(
```
Comment (Bento007) [inline backend/layers/business/business.py:132]:
```suggestion
    def get_unpublished_collection_version_from_canonical(
```

File: backend/layers/business/business.py [hunk 3]
```diff
@@ -374,39 +405,39 @@ def get_dataset_status(self, dataset_version_id: DatasetVersionId) -> DatasetSta
...
         self,
         dataset_version_id: DatasetVersionId,
-        status_key: DatasetStatusKey,
-        new_dataset_status: DatasetStatusGeneric,
+        status_key: Optional[DatasetStatusKey] = None,
+        new_dataset_status: Optional[DatasetStatusGeneric] = None,
         validation_message: Optional[str] = None,
     ) -> None:
         """
         Updates the status of a dataset version.
```
Comment (Bento007) [inline backend/layers/business/business.py:413]:
```suggestion
        TODO: split into two method, one for updating validation_message, and the other statuses.
        Updates the status of a dataset version.
```

File: backend/layers/persistence/persistence_interface.py [hunk 4]
```diff
@@ -25,7 +25,9 @@ class PersistenceException(Exception):
 
 
 class DatabaseProviderInterface:
-    def create_canonical_collection(self, owner: str, collection_metadata: CollectionMetadata) -> CollectionVersion:
+    def create_canonical_collection(
+        self, owner: str, collection_metadata: CollectionMetadata, curator_name: str
```
Comment (Bento007) [inline backend/layers/persistence/persistence_interface.py:29]: Determine where curator_name should live

File: backend/layers/persistence/persistence_mock.py [hunk 5]
```diff
@@ -67,7 +66,9 @@ def _generate_id():
     def create_canonical_collection(self, owner: str, collection_metadata: CollectionMetadata) -> CollectionVersion:
         collection_id = CollectionId(self._generate_id())
         version_id = CollectionVersionId(self._generate_id())
-        canonical = CanonicalCollection(collection_id, None, False, False)
+        canonical = CanonicalCollection(
```
Comment (Bento007) [inline backend/layers/persistence/persistence_mock.py:69]: remove and tests

File: backend/layers/persistence/persistence_mock.py [hunk 6]
```diff
@@ -80,6 +81,7 @@ def create_canonical_collection(self, owner: str, collection_metadata: Collectio
             datasets=[],
         )
         self.collections_versions[version_id.id] = version
+        self.collections[collection_id.id] = canonical
```
Comment (Bento007) [inline backend/layers/persistence/persistence_mock.py:84]: remove and tests

File: backend/portal/api/curation/v1/curation/collections/common.py [hunk 7]
```diff
@@ -42,76 +54,166 @@
...
+                collection_version.collection_id
+            )
+        revising_in = _revising_in.version_id.id if _revising_in else None
+
+    # get collection dataset attributes
+    response_datasets = []
+    collection_level_processing_status = None  # Noe if no datasets
+    for _dataset_version in collection_version.datasets:
+        if isinstance(_dataset_version, DatasetVersionId):
+            dataset_version = get_business_logic().get_dataset_version(_dataset_version)
```
Comment (Bento007) [inline backend/portal/api/curation/v1/curation/collections/common.py:124]: @ebezzi

Comment (Bento007) [reply backend/portal/api/curation/v1/curation/collections/common.py:124]: is this fork still needed?

File: tests/unit/backend/layers/business/test_business.py [hunk 8]
```diff
@@ -41,14 +40,13 @@
 from backend.layers.thirdparty.s3_provider import S3Provider
 from backend.layers.thirdparty.step_function_provider import StepFunctionProviderInterface
 from backend.layers.thirdparty.uri_provider import FileInfo, UriProviderInterface
+from unit.backend.layers.fixtures import test_user_name
```
Comment (Bento007) [inline tests/unit/backend/layers/business/test_business.py:43]:
```suggestion
from tests.unit.backend.layers.fixtures import test_user_name
```

File: tests/unit/backend/layers/common/base_api_test.py [hunk 9]
```diff
@@ -97,6 +105,7 @@ class NewBaseTest(BaseAuthAPITest):
     uri_provider: UriProviderInterface
 
     sample_dataset_metadata: DatasetMetadata
+    sample_collectoin_metadata: CollectionMetadata
```
Comment (Bento007) [inline tests/unit/backend/layers/common/base_api_test.py:108]: fix typo

File: tests/unit/backend/layers/api/test_curation_api.py [hunk 10]
```diff
@@ -0,0 +1,1229 @@
...
+        response = self.app.get(
+            f"/curation/v1/collections/test_collection_id/datasets/{bad_id}/assets",
+        )
+        self.assertEqual(404, response.status_code)
+        actual_body = response.json
+        self.assertEqual("Dataset not found.", actual_body["detail"])
+
+    def test__get_dataset_asset__collection_dataset_NOT_FOUND(self):
+        """Return Not found when the dataset is not part of the collection"""
+        bad_id = "bad_id"
```
Comment (nayib-jose-gloria) [inline tests/unit/backend/layers/api/test_curation_api.py:70]: todo: generate UUID to fit integration tests (db constraint is UUID for ID fields)

**Pull request merged**